geomstats/geometry/invariant_metric.py

```diff
@@ -363,12 +363,63 @@ def log(self, point, base_point=None):
                 point, self.group.inverse(base_point))
 
         log_from_id = self.log_from_identity(point_near_id)
-
         jacobian = self.group.jacobian_translation(
             base_point, left_or_right=self.left_or_right)
-
-        log = gs.einsum(
-            '...j,...jk->...k',
-            log_from_id,
-            Matrices.transpose(jacobian))
+        log = gs.einsum('...ij,...j->...i', jacobian, log_from_id)
         return log
+
+
+class BiInvariantMetric(InvariantMetric):
+    """Class for bi-invariant metrics which exist on Lie groups.
+
+    Compact Lie groups and direct products of compact Lie groups with vector
+    spaces admit bi-invariant metrics. Products Lie groups are not
+    implemented.
+
+    Parameters
+    ----------
+    group : LieGroup
+        The group to equip with the bi-invariant metric
+    """
+
+    def __init__(self, group):
+        super(BiInvariantMetric, self).__init__(
+            group=group, inner_product_mat_at_identity=gs.eye(group.dim),
+            default_point_type=group.default_point_type)
+        if 'SpecialOrthogonal' not in group.__str__():
```

Comment:
doesn't `isinstance` work for classes?

Reply:
Yes, but I cannot import the class SpecialOrthogonal in invariant_metric.py because special_orthogonal.py also imports invariant_metric and so it loops and doesn't work!